src/sql/azdata.proposed.d.ts

```diff
@@ -2066,6 +2066,339 @@ declare module 'azdata' {
 		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[]): Thenable<string | undefined>;
 	}
 
+	//#region Chart component model types
+
+	export type ChartType = 'bar' | 'bubble' | 'doughnut' | 'horizontalBar' | 'line' | 'pie' | 'polarArea' | 'radar' | 'scatter';
+
+	export interface ModelBuilder {
+		chart<TConfig extends ChartConfiguration>(): ComponentBuilder<ChartComponent<TConfig>, ChartComponentProperties<TConfig>>;
+	}
+
+	export interface ChartComponent<TConfig extends ChartConfiguration> extends Component, ChartComponentProperties<TConfig> {
+		onDidClick: vscode.Event<ChartClickEvent>;
+	}
+
+	export type ChartClickEvent = { label: string };
+
+	export interface ChartComponentProperties<TConfig extends ChartConfiguration> extends ComponentProperties {
+		/**
+		 * Type of chart to build
+		 * @see ChartType for a list of supported chart types
+		 */
+		chartType: ChartType;
+		/**
+		 * Configuration of chart: datasets, labels (if applicable), options
+		 */
+		configuration: TConfig;
+
+		/**
+		 * Unique ID for chart, used as the canvasID to render.
+		 */
+		chartId: string;
```

Comment:
Do we really need/want to let users configure this vs just generating our own?

Reply:
Good point; I think we can generate it.